src/devices/Ht16k33/SevenSegment.cs

```diff
@@ -0,0 +1,292 @@
+                return;
+            }
+
+            // Turn off dots
+            Dots = Dot.Off;
+
+            // Show two left dots as colon if first character is : and aligment is left
+            if (value[0] == ':' && alignment == Alignment.Left)
+            {
+                Dots |= Dot.LeftLower | Dot.LeftUpper;
+                if (value.Length == 1)
+                {
+                    WriteChars(' ', ' ', ' ', ' ');
+                    return;
+                }
+                value = value.Substring(1);
+            }
+
+            switch (value.Length)
+            {
+                case 1 when alignment == Alignment.Left:
+                    WriteChars(value[0], ' ', ' ', ' ');
+                    break;
+                case 1 when alignment == Alignment.Right:
+                    WriteChars(' ', ' ', ' ', value[0]);
+                    break;
+                case 2 when alignment == Alignment.Left:
+                    WriteChars(value[0], value[1], ' ', ' ');
+                    break;
+                case 2 when alignment == Alignment.Right:
+                    WriteChars(' ', ' ', value[0], value[1]);
+                    break;
+                case 3 when alignment == Alignment.Left && value[2] == ':':
+                    Dots |= Dot.CenterColon;
+                    WriteChars(value[0], value[1], ' ', ' ');
+                    break;
+                case 3 when alignment == Alignment.Right && value[0] == ':':
+                    Dots |= Dot.CenterColon;
+                    WriteChars(' ', ' ', value[1], value[2]);
+                    break;
+                case 3 when alignment == Alignment.Left:
+                    WriteChars(value[0], value[1], value[2], ' ');
+                    break;
+                case 3 when alignment == Alignment.Right:
+                    WriteChars(' ', value[0], value[1], value[2]);
+                    break;
+                case 4 when alignment == Alignment.Left && value[2] == ':':
+                    Dots |= Dot.CenterColon;
+                    WriteChars(value[0], value[1], value[3], ' ');
+                    break;
+                case 4 when alignment == Alignment.Right && value[1] == ':':
+                    Dots |= Dot.CenterColon;
+                    WriteChars(' ', value[0], value[2], value[3]);
+                    break;
+                case 4:
+                    WriteChars(value[0], value[1], value[2], value[3]);
+                    break;
+                case 5 when value[2] != ':':
+                    throw new ArgumentException(nameof(value), $"{nameof(value)}[2] must be a ':'");
+                case 5:
+                    Dots |= Dot.CenterColon;
+                    WriteChars(value[0], value[1], value[3], value[4]);
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(value), $"{nameof(value)} can contain maximum 5 characters");
+            }
+        }
```

Comment:
Might've gotten carried away with this WriteString helper and maybe some other Write* overloads too. Should I move these into an extension class or something?